src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,68 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { Status } from '../../types/Status';
+import { Todo } from '../../types/Todo';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as filterActions } from '../../features/filter';
+import { setTodos } from '../../features/todos';
+
+type Props = {
+  list: Todo[],
+};
+
+export const TodoFilter: React.FC<Props> = ({ list }) => {
+  const { query, status } = useAppSelector(state => state.filter);
+
+  const dispatch = useAppDispatch();
+
+  const [resetIsClicked, setResetIsClicked] = useState(false);
+
+  const selectedList = () => {
+    switch (status) {
+      case 'active':
+        return list.filter(todo => todo.completed === false);
+      case 'completed':
+        return list.filter(todo => todo.completed === true);
+      default:
+        return list;
+    }
+  };
+
+  const handleSelect: React.ChangeEventHandler<HTMLSelectElement>
+  = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    dispatch(filterActions.setFilter(value as Status));
+  };
+
+  const modifiedList = selectedList().filter(todos => {
+    return todos.title.toUpperCase().includes(query.toUpperCase());
+  });
+
+  useEffect(() => {
+    dispatch(setTodos(modifiedList));
+  }, [query, status]);
```

Comment:
you don't have to change the todos list in redux state. you can use `useMemo` hook in `TodoList` component and filter data there